firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceProcessor.java

```diff
@@ -0,0 +1,172 @@
+                    globalDevServicesConfig.timeout);
+        } catch (Throwable t) {
+            LOGGER.warn("Unable to start Firestore dev service", t);
+            // Dump captured logs in case of an error
+            compressor.closeAndDumpCaptured();
+            return null;
+        } finally {
+            compressor.close();
+        }
+
+        return devService == null ? null : devService.toBuildItem();
+    }
+
+    /**
+     * Start the container if conditions are met.
+     *
+     * @param dockerStatusBuildItem, Docker status
+     * @param config, Configuration for the Firestore service
+     * @param timeout, Optional timeout for starting the service
+     * @return Running service item, or null if the service couldn't be started
+     */
+    private DevServicesResultBuildItem.RunningDevService startContainerIfAvailable(DockerStatusBuildItem dockerStatusBuildItem,
+            FirestoreDevServiceConfig config,
+            Optional<Duration> timeout) {
+        if (!config.enabled) {
+            // Firestore service explicitly disabled
+            LOGGER.debug("Not starting Dev Services for Firestore as it has been disabled in the config");
+            return null;
+        }
+
+        if (!dockerStatusBuildItem.isDockerAvailable()) {
+            LOGGER.warn("Not starting devservice because docker is not available");
+            return null;
+        }
+
+        return startContainer(dockerStatusBuildItem, config, timeout);
+    }
+
+    /**
+     * Starts the Firestore emulator container with provided configuration.
+     *
+     * @param dockerStatusBuildItem, Docker status
+     * @param config, Configuration for the Firestore service
+     * @param timeout, Optional timeout for starting the service
+     * @return Running service item, or null if the service couldn't be started
+     */
+    private DevServicesResultBuildItem.RunningDevService startContainer(DockerStatusBuildItem dockerStatusBuildItem,
+            FirestoreDevServiceConfig config,
+            Optional<Duration> timeout) {
+        // Create and configure Firestore emulator container
+        FirestoreEmulatorContainer emulatorContainer = new QuarkusFirestoreContainer(
+                DockerImageName.parse(config.imageName).asCompatibleSubstituteFor("gcr.io/google.com/cloudsdktool/cloud-sdk"),
+                config.port.orElse(null));
+
+        // Set container startup timeout if provided
+        timeout.ifPresent(emulatorContainer::withStartupTimeout);
+        emulatorContainer.start();
+
+        // Set the config for the started container
+        FirestoreDevServiceProcessor.config = config;
+
+        // Return running service item with container details
+        return new DevServicesResultBuildItem.RunningDevService(FirestoreBuildSteps.FEATURE,
+                emulatorContainer.getContainerId(),
+                emulatorContainer::close, "quarkus.google.cloud.firestore.emulator-host",
```

Comment:
```suggestion
                emulatorContainer::close, "quarkus.google.cloud.firestore.host-override",
```

There is an existing property for that.